libs/vm/include/vm/module/binding_interfaces.hpp

```diff
@@ -37,6 +37,46 @@ namespace vm {
 
 namespace internal {
 
+template <typename T, typename Tuple>
+struct PrependedTupleImpl;
+
+template <typename T, typename... Ts>
+struct PrependedTupleImpl<T, std::tuple<Ts...>>
+{
+  using type = std::tuple<T, Ts...>;
+};
+
+template <typename T, typename Tuple>
+using PrependedTuple = typename PrependedTupleImpl<T, Tuple>::type;
```

Comment:
`Cons` would be a more traditional name here. Besides, if we take `TupleCat` from my comment above, then it is trivially `template <class Car, class Cdr> using Cons = TupleCat<std::tuple<Car>, Cdr>;`.